src/modules/CycleManager.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "../interfaces/ICycleManager.sol";
+import "../interfaces/ICycleModule.sol";
+
+/// @title CycleManager
+/// @notice Main contract that orchestrates cycle timing and transitions
+/// @dev Manages distribution cycles and ensures predictable, deterministic execution
+contract CycleManager is ICycleManager {
+    /// @notice The cycle module implementation
+    ICycleModule public cycleModule;
+
+    /// @notice Addresses authorized to trigger cycle transitions
+    mapping(address => bool) public authorized;
+
+    /// @notice Tracks whether distribution has been completed for the current cycle
+    bool public distributionCompletedForCurrentCycle;
+
+    /// @notice Error thrown when caller is not authorized
+    error NotAuthorized();
+
+    /// @notice Error thrown when cycle module is not set
+    error CycleModuleNotSet();
+
+    /// @notice Error thrown when cycle transition is invalid
+    error InvalidCycleTransition();
+
+    /// @notice Error thrown when attempting to start a new cycle before distribution is completed
+    error DistributionNotCompleted();
+
+    /// @notice Error thrown when cycle module address is invalid
+    error InvalidCycleModuleAddress();
+    /// @notice Emitted when an address authorization status changes
+    /// @param account The address whose authorization was updated
+    /// @param isAuthorized The new authorization status
+
+    event AuthorizationUpdated(address indexed account, bool isAuthorized);
+
+    /// @notice Initializes the contract and authorizes the deployer
+    constructor() {
+        authorized[msg.sender] = true;
+        emit AuthorizationUpdated(msg.sender, true);
+    }
+
+    /// @notice Modifier to restrict access to authorized addresses
+    modifier onlyAuthorized() {
+        if (!authorized[msg.sender]) {
+            revert NotAuthorized();
+        }
+        _;
+    }
+
+    /// @notice Modifier to ensure cycle module is set
+    modifier cycleModuleSet() {
+        if (address(cycleModule) == address(0)) {
+            revert CycleModuleNotSet();
+        }
+        _;
+    }
```

Comment:
if this is just set in the constructor we don't have to have this modifier and checks